oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverProvider.java

```diff
@@ -66,22 +66,31 @@ public void start() throws ServiceNotProvidedException, ModuleStartException {
             throw new IllegalArgumentException(
                 "sampleRate: " + config.getSampleRate() + ", should be between 0 and 10000");
         }
-        HTTPServerConfig httpServerConfig = HTTPServerConfig.builder()
-                                                            .host(config.getRestHost())
-                                                            .port(config.getRestPort())
-                                                            .contextPath(config.getRestContextPath())
-                                                            .idleTimeOut(config.getRestIdleTimeOut())
-                                                            .maxThreads(config.getRestMaxThreads())
-                                                            .acceptQueueSize(config.getRestAcceptQueueSize())
-                                                            .build();
 
-        httpServer = new HTTPServer(httpServerConfig);
-        httpServer.initialize();
+        if (config.isEnableHttpCollector()) {
+            HTTPServerConfig httpServerConfig = HTTPServerConfig.builder()
+                                                                .host(config.getRestHost())
+                                                                .port(config.getRestPort())
+                                                                .contextPath(config.getRestContextPath())
+                                                                .idleTimeOut(config.getRestIdleTimeOut())
+                                                                .maxThreads(config.getRestMaxThreads())
+                                                                .acceptQueueSize(config.getRestAcceptQueueSize())
+                                                                .build();
+
+            httpServer = new HTTPServer(httpServerConfig);
```

Comment:
Putting this in condition `if (config.isEnableHttpCollector())` might cause NPE at line 98 when the condition is `false`

```java

    @Override
    public void notifyAfterCompleted() {
        httpServer.start();
    }
```

Reply:
fixed